### PR TITLE
1893: Corrections for errors discovered during pre-alpha testing

### PR DESCRIPTION
**Pull request by perwestling**
Corrections:
- Add AdSK to liquidity calculation
- Fix several nil calls
- Corrected par calculation when using FdSD to par
- Corrected value for MR (same as round before, so do not appear
  in spread sheet) including not moving share prices after MR
- Corrected KV269 terrain cost ("Gray phase" variant)
- Close minors that are part of AGV/HGK formation
- Corrected FdSD exchange
- Corrected logging of treasury increase during merge
- Corrected merge round entry during phase 6 (auto form HGK)
- Fixed so that Leverkusen can be upgraded to green

Improvements:
- Improved name and description for minors
- Links to rule and rule summary
- MR is not shown after HGK has been formed
- Added detailed description of AdSK, AGV, HGK abilites on charter
- J5 token now starts as an placed HGK token from the start
- Added a pre-alpha game as a spec file to ensure regression testing
- Show the charters of merger involved minors

### CODE REVIEW

File: lib/engine/game/g_1893/corporation.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+require_relative '../../corporation'
+
+module Engine
+  module Game
+    module G1893
+      class Corporation < Engine::Corporation
+        def floated?
+          @name == 'AdSK' || super
```
Comment (tobymao) [inline lib/engine/game/g_1893/corporation.rb:10]: what do you mean that adsk is bonds? what does that mean?

File: lib/engine/game/g_1893/game.rb [hunk 2]
```diff
@@ -828,16 +883,22 @@ def operating_round(round_num)
...
             Engine::Step::Token,
             Engine::Step::Route,
             G1893::Step::Dividend,
             G1893::Step::BuyTrain,
           ], round_num: round_num)
         end
 
+        def operating_order
+          # AdSK is bonds, and do not operate
+          super.reject { |c| c == adsk }
```
Comment (tobymao) [inline lib/engine/game/g_1893/game.rb:896]: then maybe adsk shouldn't be a corp?

File: lib/engine/game/g_1893/minor.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require_relative '../../minor'
+
+module Engine
+  module Game
+    module G1893
+      class Minor < Engine::Minor
```
Comment (tobymao) [inline lib/engine/game/g_1893/minor.rb:8]: minors don't have shares can we delete this class

File: lib/engine/game/g_1893/share_pool.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require_relative 'share_pool'
+
+module Engine
+  module Game
+    module G1893
+      class SharePool < Engine::SharePool
```
Comment (tobymao) [inline lib/engine/game/g_1893/share_pool.rb:8]: again the problem is adsk, so delete this class as well

File: lib/engine/game/g_1893/step/track.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            end.compact
+          end
+
+          def leverkusen_only_legal_rotation(tile)
+            tile.rotate!(0)
+            tile
+          end
+
+          def legal_tile_rotation?(entity, hex, tile)
+            return false unless @game.legal_tile_rotation?(entity, hex, tile)
```
Comment (tobymao) [inline lib/engine/game/g_1893/step/track.rb:31]: yea let's delete this

File: lib/engine/game/g_1893/corporation.rb (hunk 1 above)
Comment (perwestling) [reply lib/engine/game/g_1893/corporation.rb]: AdSK is the short name of the bonds used in 1893. They have a set price and revenue, and is bought/sold as shares.

File: lib/engine/game/g_1893/game.rb (hunk 2 above)
Comment (perwestling) [reply lib/engine/game/g_1893/game.rb]: Maybe not, but making it a company is not a perfect fit, either. 

Probably there should be a bond class (or similar).

But I am trying it as a company in the latest version.

File: lib/engine/game/g_1893/minor.rb (hunk 3 above)
Comment (perwestling) [reply lib/engine/game/g_1893/minor.rb]: Has been deleted.

File: lib/engine/game/g_1893/share_pool.rb (hunk 4 above)
Comment (perwestling) [reply lib/engine/game/g_1893/share_pool.rb]: As mentioned above, making AdsK company created other problems.

But yes, it has been deleted.

File: lib/engine/game/g_1893/step/track.rb (hunk 5 above)
Comment (perwestling) [reply lib/engine/game/g_1893/step/track.rb]: Improved solution now used.

**Comment by perwestling** [comment]
@tobymao Now the commit is ready to be reviewed.
Have changed how AdSK and minors are handled.

File: lib/engine/game/g_1893/game.rb [hunk 6]
```diff
@@ -734,6 +649,80 @@ def remove_tiles(tiles)
           end
         end
 
+        def game_companies
+          all_companies = [
```
Comment (tobymao) [inline lib/engine/game/g_1893/game.rb:653]: constant

File: lib/engine/game/g_1893/game.rb [hunk 7]
```diff
@@ -734,6 +649,80 @@ def remove_tiles(tiles)
...
+            all_companies << {
+              sym: "AdSK-#{x}",
+              name: "Anleihen der Stadt Köln #{x}",
+              value: 100,
+              revenue: 10,
+              desc: 'Bond. Can be bought/sold during Stock Round in a similar way shares are bought/sold.',
+              color: nil,
+            }
+          end
+          all_companies
```
Comment (tobymao) [inline lib/engine/game/g_1893/game.rb:710]: COMPANIES + (1..10).map do...

**Comment by tobymao** [comment]
looks good

**Comment by perwestling** [comment]
@tobymao - have now fixed your review comments, and moved it from Draft status. 
So it is ready to deliver if it is approved.

**Pull request merged**